code/GridFieldArbitraryData/ArbitraryDataAdmin.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+namespace SilverStripe\FrameworkTest\GridFieldArbitraryData;
+
+use RuntimeException;
+use SilverStripe\Admin\LeftAndMain;
+use SilverStripe\Control\Controller;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridFieldConfig;
+use SilverStripe\Forms\GridField\GridFieldConfig_Base;
+use SilverStripe\Forms\GridField\GridFieldConfig_RecordEditor;
+use SilverStripe\Forms\GridField\GridFieldDataColumns;
+use SilverStripe\Forms\GridField\GridFieldDetailForm;
+use SilverStripe\Forms\GridField\GridFieldExportButton;
+use SilverStripe\Forms\GridField\GridFieldFilterHeader;
+use SilverStripe\Forms\GridField\GridFieldPaginator;
+use SilverStripe\Forms\GridField\GridFieldPrintButton;
+use SilverStripe\Forms\GridField\GridFieldViewButton;
+use SilverStripe\Forms\HiddenField;
+use SilverStripe\Forms\TextField;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\Queries\SQLSelect;
+use SilverStripe\ORM\Search\BasicSearchContext;
+use SilverStripe\View\ArrayData;
+
+class ArbitraryDataAdmin extends LeftAndMain
```

Comment:
Might be worthwhile exploring if we want to provide something like this in core or in an optional module.